pkg/diffs/diffs_test.go

```diff
@@ -3,14 +3,88 @@ package diffs
 import (
 	"testing"
 
+	"github.com/getlantern/deepcopy"
 	"github.com/sirupsen/logrus"
+
 	"k8s.io/api/core/v1"
+
 	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/util/diff"
+
 	prowconfig "k8s.io/test-infra/prow/config"
 
-	"github.com/getlantern/deepcopy"
+	cioperatorapi "github.com/openshift/ci-operator/pkg/api"
+
+	"github.com/openshift/ci-operator-prowgen/pkg/config"
 )
 
+func TestGetChangedCiopConfigs(t *testing.T) {
+	baseCiopConfig := cioperatorapi.ReleaseBuildConfiguration{
+		InputConfiguration: cioperatorapi.InputConfiguration{
+			ReleaseTagConfiguration: &cioperatorapi.ReleaseTagConfiguration{
+				Cluster:   "kluster",
+				Namespace: "namespace",
+				Tag:       "tag",
+			},
+		},
+	}
+
+	testCases := []struct {
+		name            string
+		configGenerator func() (before, after config.CompoundCiopConfig)
+		expected        func() config.CompoundCiopConfig
+	}{{
+		name: "no changes",
+		configGenerator: func() (config.CompoundCiopConfig, config.CompoundCiopConfig) {
+			before := config.CompoundCiopConfig{"org-repo-branch.yaml": &baseCiopConfig}
+			after := config.CompoundCiopConfig{"org-repo-branch.yaml": &baseCiopConfig}
+			return before, after
+		},
+		expected: func() config.CompoundCiopConfig { return config.CompoundCiopConfig{} },
+	}, {
+		name: "new config",
+		configGenerator: func() (config.CompoundCiopConfig, config.CompoundCiopConfig) {
+			before := config.CompoundCiopConfig{"org-repo-branch.yaml": &baseCiopConfig}
+			after := config.CompoundCiopConfig{
+				"org-repo-branch.yaml":         &baseCiopConfig,
+				"org-repo-another-branch.yaml": &baseCiopConfig,
+			}
+			return before, after
+		},
+		expected: func() config.CompoundCiopConfig {
+			return config.CompoundCiopConfig{"org-repo-another-branch.yaml": &baseCiopConfig}
+		},
+	}, {
+		name: "changed config",
+		configGenerator: func() (config.CompoundCiopConfig, config.CompoundCiopConfig) {
+			before := config.CompoundCiopConfig{"org-repo-branch.yaml": &baseCiopConfig}
+			afterConfig := cioperatorapi.ReleaseBuildConfiguration{}
+			deepcopy.Copy(&afterConfig, baseCiopConfig)
+			afterConfig.InputConfiguration.ReleaseTagConfiguration.Tag = "another-tag"
+			after := config.CompoundCiopConfig{"org-repo-branch.yaml": &afterConfig}
+			return before, after
+		},
+		expected: func() config.CompoundCiopConfig {
+			expected := cioperatorapi.ReleaseBuildConfiguration{}
+			deepcopy.Copy(&expected, baseCiopConfig)
+			expected.InputConfiguration.ReleaseTagConfiguration.Tag = "another-tag"
+			return config.CompoundCiopConfig{"org-repo-branch.yaml": &expected}
+		},
+	}}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			before, after := tc.configGenerator()
+			actual := GetChangedCiopConfigs(before, after, logrus.NewEntry(logrus.New()))
+			expected := tc.expected()
+
+			if !equality.Semantic.DeepEqual(expected, actual) {
```

Comment:
CI op configs are not k8s objects, should use reflect.DeepEqual?

Reply:
I guess we should really use `reflect` here. Ordering should not matter as `CompoundConfig` is a map.

Reply:
I mean the value of a specific key. But we should not care about that in the tests. So I agree to use `reflect` here.